dissect/target/plugins/os/unix/services.py

```diff
@@ -1,86 +1,147 @@
 import re
+from itertools import chain
+from typing import BinaryIO, Iterator
 
 from dissect.target.exceptions import UnsupportedPluginError
 from dissect.target.helpers.record import TargetRecordDescriptor
-from dissect.target.plugin import Plugin, export
+from dissect.target.plugin import Plugin, export, internal
 
 LinuxServiceRecord = TargetRecordDescriptor(
     "linux/service",
     [
         ("datetime", "ts"),
         ("string", "name"),
-        ("string", "servicepath"),
-        ("string", "servicevariables"),
+        ("string", "config"),
+        ("path", "source"),
     ],
 )
 
 
-class ServicePlugin(Plugin):
-    PATHS = (
+class ServicesPlugin(Plugin):
+    SYSTEMD_PATHS = [
         "/etc/systemd/system",
         "/lib/systemd/system",
         "/usr/lib/systemd/system",
-    )
+    ]
+
+    INITD_PATHS = ["/etc/rc.d/init.d", "/etc/init.d"]
 
     def check_compatible(self):
-        if (
-            not any([self.target.fs.path(p).exists() for p in self.PATHS])
-            and not self.target.fs.path("/etc/init.d").exists()
-        ):
+        if not any([self.target.fs.path(p).exists() for p in self.SYSTEMD_PATHS + self.INITD_PATHS]):
             raise UnsupportedPluginError("No supported service directories found")
 
     @export(record=LinuxServiceRecord)
-    def services(self):
-        """Return information about all installed services."""
+    def services(self) -> Iterator[LinuxServiceRecord]:
+        """Return information about all installed systemd and init.d services.
+
+        Resources:
```

Comment:
```suggestion
        References:
```

We recently changed the wording. This is the keyword the sphinx doc generator recognizes.